backend/src/agent/persistence/worker/metrics_reporter.go

```diff
@@ -112,5 +112,5 @@ func aggregateErrors(errors []error) error {
 		}
 		errorMsgs = append(errorMsgs, err.Error())
 	}
-	return util.NewCustomErrorf(code, strings.Join(errorMsgs, "\n"))
+	return util.NewCustomErrorf(code, "%s", strings.Join(errorMsgs, "\n"))
```

Comment:
Why is this and other similar changes needed? Doesn't the underlying call of `fmt.Sprintf(format, a...)` support `a` being empty?